File: javascript/08-jealous-function/test/kitchen_ticket.test.js
import assert from 'node:assert';
import test from 'node:test';
import { KitchenTicket, Order } from '../src/kitchen_ticket.js';

function makeOrder({ items, customer, table, special = '', rush = false } = {}) {
  return new Order(
    items || [{ name: 'Burger', qty: 1 }],
    customer || { name: 'Alice', vip: false },
    table || { number: 5, zone: 'patio', server: 'Sam' },
    special,
    rush,
  );
}

test('prints table customer and server details', () => {
  const result = new KitchenTicket().print_ticket(makeOrder());
  assert.ok(result.includes('Table: 5'));
  assert.ok(result.includes('Zone: patio'));
  assert.ok(result.includes('Server: Sam'));
  assert.ok(result.includes('Customer: Alice'));
});

test('prints ticket with multiple items and count', () => {
  const result = new KitchenTicket().print_ticket(makeOrder({ items: [{ name: 'Burger', qty: 2 }, { name: 'Fries', qty: 1 }] }));
  assert.ok(result.includes('Burger x2'));
  assert.ok(result.includes('Fries x1'));
  assert.ok(result.includes('Items: 3'));
});

test('prints modifiers and allergy flags', () => {
  const result = new KitchenTicket().print_ticket(makeOrder({ items: [{ name: 'Salad', qty: 1, modifiers: ['no onion', 'dressing side'], allergy: 'nuts' }] }));
  assert.ok(result.includes('Salad x1 [no onion, dressing side] ALLERGY:nuts'));
});

test('prints vip and rush markers', () => {
  const result = new KitchenTicket().print_ticket(makeOrder({ customer: { name: 'Carol', vip: true }, rush: true }));
  assert.ok(result.includes('VIP'));
  assert.ok(result.includes('RUSH'));
});

test('omits special when empty but keeps separator', () => {
  const result = new KitchenTicket().print_ticket(makeOrder({ special: '' }));
  assert.ok(!result.includes('Special:'));
  assert.ok(result.includes('---'));
});

test('prints special instructions', () => {
  const result = new KitchenTicket().print_ticket(makeOrder({ special: 'Fire mains after starters' }));
  assert.ok(result.includes('Special: Fire mains after starters'));
});

File: csharp/08-jealous-function/test/kitchen_ticket_test.cs
using Xunit;

public class KitchenTicketTest
{
    private KitchenOrder Order(
        List<TicketItem>? items = null,
        Dictionary<string, object>? customer = null,
        Dictionary<string, object>? table = null,
        string special = "",
        bool rush = false)
    {
        return new KitchenOrder(
            items ?? new List<TicketItem> { new TicketItem("Burger", 1) },
            customer ?? new Dictionary<string, object> { { "name", "Alice" }, { "vip", false } },
            table ?? new Dictionary<string, object> { { "number", 5 }, { "zone", "patio" }, { "server", "Sam" } },
            special,
            rush);
    }

    [Fact]
    public void PrintsTableCustomerAndServerDetails()
    {
        var result = new KitchenTicket().print_ticket(Order());
        Assert.Contains("Table: 5", result);
        Assert.Contains("Zone: patio", result);
        Assert.Contains("Server: Sam", result);
        Assert.Contains("Customer: Alice", result);
    }

    [Fact]
    public void PrintsTicketWithMultipleItemsAndCount()
    {
        var result = new KitchenTicket().print_ticket(Order(items: new List<TicketItem> { new TicketItem("Burger", 2), new TicketItem("Fries", 1) }));
        Assert.Contains("Burger x2", result);
        Assert.Contains("Fries x1", result);
        Assert.Contains("Items: 3", result);
    }

    [Fact]
    public void PrintsModifiersAndAllergyFlags()
    {
        var result = new KitchenTicket().print_ticket(Order(items: new List<TicketItem> { new TicketItem("Salad", 1, new List<string> { "no onion", "dressing side" }, "nuts") }));
        Assert.Contains("Salad x1 [no onion, dressing side] ALLERGY:nuts", result);
    }

    [Fact]
    public void PrintsVipAndRushMarkers()
    {
        var result = new KitchenTicket().print_ticket(Order(customer: new Dictionary<string, object> { { "name", "Carol" }, { "vip", true } }, rush: true));
        Assert.Contains("VIP", result);
        Assert.Contains("RUSH", result);
    }

    [Fact]
    public void OmitsSpecialWhenEmptyButKeepsSeparator()
    {
        var result = new KitchenTicket().print_ticket(Order(special: ""));
        Assert.DoesNotContain("Special:", result);
        Assert.Contains("---", result);
    }

    [Fact]
    public void PrintsSpecialInstructions()
    {
        var result = new KitchenTicket().print_ticket(Order(special: "Fire mains after starters"));
        Assert.Contains("Special: Fire mains after starters", result);
    }
}
